internal/service/logs/anomaly_detector.go

```diff
@@ -0,0 +1,299 @@
+	"github.com/hashicorp/terraform-plugin-framework/path"
+	"github.com/hashicorp/terraform-plugin-framework/resource"
+	"github.com/hashicorp/terraform-plugin-framework/resource/schema"
+	"github.com/hashicorp/terraform-plugin-framework/resource/schema/int64planmodifier"
+	"github.com/hashicorp/terraform-plugin-framework/resource/schema/planmodifier"
+	"github.com/hashicorp/terraform-plugin-framework/resource/schema/stringplanmodifier"
+	"github.com/hashicorp/terraform-plugin-framework/schema/validator"
+	"github.com/hashicorp/terraform-plugin-framework/types"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/retry"
+	"github.com/hashicorp/terraform-provider-aws/internal/create"
+	"github.com/hashicorp/terraform-provider-aws/internal/enum"
+	"github.com/hashicorp/terraform-provider-aws/internal/errs"
+	"github.com/hashicorp/terraform-provider-aws/internal/framework"
+	"github.com/hashicorp/terraform-provider-aws/internal/framework/flex"
+	fwtypes "github.com/hashicorp/terraform-provider-aws/internal/framework/types"
+	tftags "github.com/hashicorp/terraform-provider-aws/internal/tags"
+	"github.com/hashicorp/terraform-provider-aws/internal/tfresource"
+	"github.com/hashicorp/terraform-provider-aws/names"
+)
+
+// @FrameworkResource("aws_cloudwatch_log_anomaly_detector", name="Anomaly Detector")
+// @Tags(identifierAttribute="arn")
+// @Testing(importStateIdFunc="testAccAnomalyDetectorImportStateIDFunc")
+// @Testing(importStateIdAttribute="arn")
+// @Testing(importIgnore="enabled")
+// @Testing(existsType="github.com/aws/aws-sdk-go-v2/service/logs;cloudwatchlogs.GetLogAnomalyDetectorOutput")
+func newResourceAnomalyDetector(_ context.Context) (resource.ResourceWithConfigure, error) {
+	r := &resourceAnomalyDetector{}
+
+	return r, nil
+}
+
+const (
+	ResNameAnomalyDetector = "Anomaly Detector"
+)
+
+type resourceAnomalyDetector struct {
+	framework.ResourceWithConfigure
+}
+
+func (r *resourceAnomalyDetector) Metadata(_ context.Context, req resource.MetadataRequest, resp *resource.MetadataResponse) {
+	resp.TypeName = "aws_cloudwatch_log_anomaly_detector"
+}
+
+func (r *resourceAnomalyDetector) Schema(ctx context.Context, req resource.SchemaRequest, resp *resource.SchemaResponse) {
+	resp.Schema = schema.Schema{
+		Attributes: map[string]schema.Attribute{
+			names.AttrARN: framework.ARNAttributeComputedOnly(),
+			"log_group_arn_list": schema.ListAttribute{
+				CustomType:  fwtypes.ListOfStringType,
+				ElementType: types.StringType,
+				Required:    true,
+			},
+			"anomaly_visibility_time": schema.Int64Attribute{
+				Optional: true,
+				Computed: true,
+				Validators: []validator.Int64{
+					int64validator.AtLeast(7),
+					int64validator.AtMost(90),
+				},
+				PlanModifiers: []planmodifier.Int64{
+					int64planmodifier.UseStateForUnknown(),
+				},
+			},
+			"detector_name": schema.StringAttribute{
+				Optional: true,
+			},
+			"evaluation_frequency": schema.StringAttribute{
+				Optional: true,
```

Comment:
Using the enum custom type here lest you remove the validator

```suggestion
                                 CustomType: fwtypes.StringEnumType[awstypes.EvaluationFrequency](),
				Optional: true,
```